### ISSUE TITLE
Test cubeviz base functionality in matplotlib + ipywidgets

### ISSUE THREAD

**Issue by SaOgaz** [spacetelescope/dat_pyinthesky#22]
From [functional spec](https://innerspace.stsci.edu/display/DSMO/Elaboration%3A+Cubeviz-in-browser+prototype?focusedCommentId=162048038#comment-162048038), do 1-3 using matplotlib + ipywidgets. Stetch goal to achieve 4 (displaying in a separate ipywidgets text box but with cursor information from matplotlib).

Wording of tasks (1-3) may be re-done in the near future, so keep that in mind.  

Please feel free to post any and all implementation questions here in the issue!  This will be the best recorded forum to get any details worked out that might come up.

JIRA product board ticket: https://jira.stsci.edu/browse/DATJPP-12

**Comment by ibusko** [comment]
There is a new notebook, dat_pyinthesky_test/jwst_nb_viz/example/cubeviz-ipywidgets.ipynb which exercises the 1-3 requirements. It is mostly based on an existing notebook in dat_pyinthesky_test/jwst_nb_viz/cubehack-ipywidgets, which already performs similar operations to the ones we need. So I used it as template code. 

This seems OK for a first iteration, but more work is needed if we want to pursue this path (ipywidgets+matplotlib).

**Comment by ibusko** [comment]
I did a lot of cleanup on the code in notebook `(ibusko):dat_pyinthesky_test/jwst_nb_viz/example/cubeviz-ipywidgets.ipynb`. I solved most of the problems described in above comments. 

The one remaining feature not yet implemented is interactivity of the 1D plot. The plot should respond to where on the image stack the cursor is poiting so as to plot the spectrum corresponding to the pointed image pixel.
